packages/core/src/shipping/strategies/stripe-upe/stripe-upe-shipping-strategy.ts

```diff
@@ -0,0 +1,113 @@
+import { PaymentMethodActionCreator } from '../../../payment';
+
+import { AddressRequestBody } from '../../../address';
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { StripeElements, StripeElementType, StripeEventType, StripeScriptLoader, StripeUPEAppearanceOptions, StripeUPEClient } from '../../../payment/strategies/stripe-upe';
+import ConsignmentActionCreator from '../../consignment-action-creator';
+import { ShippingInitializeOptions, ShippingRequestOptions } from '../../shipping-request-options';
+import ShippingStrategy from '../shipping-strategy';
+
+export default class StripeUPEShippingStrategy implements ShippingStrategy {
+    private _stripeUPEClient?: StripeUPEClient;
+    private _stripeElements?: StripeElements;
+    private sendData?: ReturnType<typeof setTimeout>;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _stripeUPEScriptLoader: StripeScriptLoader,
+        private _consignmentActionCreator: ConsignmentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator
+    ) {}
+
+    updateAddress(address: AddressRequestBody, options?: ShippingRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(
+            this._consignmentActionCreator.updateAddress(address, options)
+        );
+    }
+
+    selectOption(optionId: string, options?: ShippingRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(
+            this._consignmentActionCreator.selectShippingOption(optionId, options)
+        );
+    }
+
+    async initialize(options: ShippingInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId = "", stripeupe: { container, onChangeShipping, getStyles, availableCountries } = {} } = options;
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId, { params: { method: 'card' } }));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow('card', methodId);
```

Comment:
Same as above

Reply:
@animesh1987 sorry, I don't understand the change here because It's the same file, just in case I added the change on the `stripe-upe-customer-strategy.ts` on the commit called `feat(payment): STRIPE-52 animesh & peng a proposal to solve the feedback`